src/control/server/ownership.go

```diff
@@ -135,3 +131,32 @@ func changeFileOwnership(config *configuration) error {
 
 	return nil
 }
+
+// respawnProc builds and executes command chain to respawn process under new ownership.
+func respawnProc(config *configuration) error {
+	var buf bytes.Buffer // build command string
+
+	// Wait for this proc to exit and make NVMe storage accessible to new user.
+	fmt.Fprintf(
+		&buf, `sleep 1 && %s storage prep-nvme -u %s &> %s`,
+		os.Args[0], config.UserName, config.ControlLogFile)
+
+	// Run daos_server from within a subshell of target user with the same args.
+	fmt.Fprintf(&buf, ` && su %s -c "`, config.UserName)
+
+	for _, arg := range os.Args {
```

Comment:
you could also do this: 
```go
fmt.Fprintf(&buf, "%s", strings.Join(os.Args))
```
or, probably better, if you use strings.Builder you could simplify a lot of stuff here and avoid using fmt.Fprintf:

```go
buf.WriteString(strings.Join(os.Args, " "))

Reply:
in the other cases, is `buf.WriteString(fmt.Sprintf(` && su %s -c "`, config.UserName))` simpler than `fmt.Fprintf(&buf, ` && su %s -c "`, config.UserName)` or is there another way of doing it?